src/vdaf.rs

```diff
@@ -279,23 +263,23 @@ impl<F> AsRef<[F]> for OutputShare<F> {
     }
 }
 
-impl<F> From<Vec<F>> for OutputShare<F> {
+impl<F: FieldElement> From<Vec<F>> for OutputShare<F> {
```

Comment:
AFAICT nothing in this `impl` depends on `F` being `FieldElement` so there's no need to constrain `F` at all.

Reply:
Ah, I should have commented: this is required as otherwise it conflicts with the `TryFrom<Vec<u8>>` implementation -- specifically, I think what's going on is that, letting `F = u8`, we have a `From<Vec<u8>>` implementation here, which implies a `TryFrom<Vec<u8>>` implementation, which conflicts with the manual implementation below. By restricting `F` to `FieldElement` we resolve the conflict, since `u8` does not implement `FieldElement`.

I think this is an OK change as, despite being a general container type, `OutputShare`/`AggregateShare` are will always store some `FieldElement` in practice.